src/wp-includes/media.php

```diff
@@ -5456,11 +5460,34 @@ function wp_get_loading_attr_default( $context ) {
 		return false;
 	}
 
+	// Conditionally skip lazy-loading on images before the loop.
+	static $special_content_contexts = array(
+		'wp_get_attachment_image' => true,
+		'the_post_thumbnail'      => true,
+	);
+	if (
+		isset( $special_content_contexts[ $context ] )
+		// Only apply for main query but before the loop.
+		&& $wp_query->before_loop && is_main_query()
+		/*
+		 * Any image before the loop, but after the header has started should not be lazy-loaded,
+		 * except when the footer has already started which can happen when the current template
+		 * does not include any loop.
+		 */
+		&& did_action( 'get_header' ) && ! did_action( 'get_footer' )
+	) {
+		return false;
+	}
+
 	/*
 	 * The first elements in 'the_content' or 'the_post_thumbnail' should not be lazy-loaded,
 	 * as they are likely above the fold.
 	 */
-	if ( 'the_content' === $context || 'the_post_thumbnail' === $context ) {
+	static $content_contexts = array(
+		'the_content'        => true,
+		'the_post_thumbnail' => true,
+	);
+	if ( isset( $content_contexts[ $context ] ) ) {
```

Comment:
While maybe this is a good idea, let's keep this code like it was before as it has nothing to do with the actual change this PR is for.